dlib/control/approximate_linear_models.h

```diff
@@ -56,6 +58,50 @@ namespace dlib
 // ----------------------------------------------------------------------------------------
 
     template <
+            typename feature_extractor
+            >
+    struct extended_process_sample : public process_sample<feature_extractor>
```

Comment:
Please don't use inheritance here.  Only use inheritance when it makes sense to treat an object polymorphically.  But you would never want to pass an extended process sample to something expecting a process sample, or at least it would surprise me and I doubt it's what a user would really want to do.  

The idea of having an extended process sample is good though.  But make it a separate object from process_sample.

Reply:
Here I used inheritance with the intention of being able of using the three algorithms with the same code. You can see an example in the test code where I create extended_processes and by slicing they also work on qlearning & lspi.

I think whether it makes sense to use inheritance or not comes to whether people likely want to use those algorithms at the same time. Your right tho, in this case it might be better to not use it.

Reply:
Well, maybe you are right.  I'm on the fence about it after thinking a bit more.  If you want to leave it this way then that's cool with me :)